src/OgAccess.php

```diff
@@ -0,0 +1,202 @@
+   *   hook_og_user_access_alter() that still want to use og_user_access(), but
+   *   without causing a recursion. Defaults to FALSE.
+   * @param $ignore_admin
+   *   (optional) When TRUE the specific permission is checked, ignoring the
+   *   "administer group" permission if the user has it. When FALSE, a user
+   *   with "administer group" will be granted all permissions.
+   *   Defaults to FALSE.
+   *
+   * @return bool
+   *   TRUE or FALSE if the current user has the requested permission.
+   *   NULL, if the given group isn't a valid group.
+   */
+  public static function userAccess($group_type, $group_id, $operation, $account = NULL, $skip_alter = FALSE, $ignore_admin = FALSE) {
+    $perm = &drupal_static(__FUNCTION__, []);
+    // Mark the group ID and permissions that invoked an alter.
+    $perm_alter = &drupal_static(__FUNCTION__ . '_alter', []);
+
+    if (!Og::isGroup($group_type, $group_id)) {
+      // Not a group.
+      return NULL;
+    }
+
+
+    if (empty($account)) {
+      $account = clone \Drupal::currentUser()->getAccount();
+    }
+
+    $account_id = $account->id();
+
+    // User #1 has all privileges.
+    if ($account_id == 1) {
+      return TRUE;
+    }
+
+    // Administer group permission.
+    if (!$ignore_admin && $account->hasPermission(static::ADMINISTER_GROUP_PERMISSION)) {
+      return TRUE;
+    }
+
+    // Group manager has all privileges (if variable is TRUE).
+    if (!empty($account_id) && \Drupal::config('og.settings')->get('group_manager_full_access')) {
+      $group = entity_load($group_type, $group_id);
+
+      if (($group instanceof EntityOwnerInterface) && ($group->getOwnerId() == $account_id)) {
+        return TRUE;
+      }
+    }
+
+    $identifier = $group_type . ':' . $group_id;
+
+    // To reduce the number of SQL queries, we cache the user's permissions
+    // in a static variable.
+    if (!isset($perm[$identifier][$account_id])) {
+      $perms = array();
+
+      if ($roles = og_get_user_roles($group_type, $group_id, $account_id)) {
+        // Member might not have roles if they are blocked.
+        // A pending member is treated as a non-member.
+        $role_permissions = og_role_permissions($roles);
+
+        foreach ($role_permissions as $one_role) {
+          $perms += $one_role;
+        }
+      }
+
+      $perm[$identifier][$account_id] = $perms;
+    }
+
+    if (!$skip_alter && empty($perm_alter[$identifier][$account_id][$operation])) {
+      // Let modules alter the permissions. since $perm is static we create
+      // a clone of it.
+      $group = !empty($group) ? $group : entity_load($group_type, $group_id);
+      $temp_perm = $perm[$identifier][$account_id];
+      $context = array(
+        'operation' => $operation,
+        'group_type' => $group_type,
+        'group' => $group,
+        'account' => $account,
+      );
+
+      drupal_alter('og_user_access', $temp_perm, $context);
```

Comment:
I think this is good as an alter hook still. It is not something you would want multiple instances of etc.. Plugins get a bit weird when used outside their usual patterns. Just look at the element plugins in core.... :)

I will convert that to use the module handler directly though.